### PR TITLE
Detail view layout

### PR DESCRIPTION
**Pull request by SMahil**
Fix Css for Detail Page Layout. It comprises of following components
- Detail View Header
- Total Links Component
- Assignees, Creator, Iterations Wrapper
- Labels Dropdown

Issue:
https://github.com/openshiftio/openshift.io/issues/963
https://github.com/openshiftio/openshift.io/issues/964 (Part of it has been resolved)
https://openshift.io/openshiftio/openshiftio/plan/detail/1551
https://openshift.io/openshiftio/openshiftio/plan/detail/1547

### CODE REVIEW

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2177-2 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2177-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2177-1 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2177-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2177-3 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2177-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2177-4 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2177-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2177-5 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2177-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2177-7 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2177-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2177-8 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2177-fabric8-planner.badger.fabric8.io

File: src/app/components/work-item-new-detail/work-item-new-detail.component.ts [hunk 1]
```diff
@@ -141,6 +148,17 @@ export class WorkItemNewDetailComponent implements OnInit, OnDestroy {
     this.eventListeners.forEach(subscriber => subscriber.unsubscribe());
   }
 
+  ngDoCheck() {
+    if(this.detailHeader){
+      let HdrDivHeight :any =  this.detailHeader.nativeElement.offsetHeight;
+      let targetHeight :any = window.innerHeight - HdrDivHeight - 90;
```
Comment (vikram-raj) [inline src/app/components/work-item-new-detail/work-item-new-detail.component.ts]: No need to give space before :

File: src/app/components/work-item-new-detail/work-item-new-detail.component.ts [hunk 2]
```diff
@@ -141,6 +148,17 @@ export class WorkItemNewDetailComponent implements OnInit, OnDestroy {
     this.eventListeners.forEach(subscriber => subscriber.unsubscribe());
   }
 
+  ngDoCheck() {
+    if(this.detailHeader){
```
Comment (vikram-raj) [inline src/app/components/work-item-new-detail/work-item-new-detail.component.ts]: Add space before {

File: src/app/widgets/inlineinput/inlineinput.component.less [hunk 3]
```diff
@@ -1,60 +1,60 @@
...
+    background-color: @color-pf-white;
+    .Border(1px, solid, @color-pf-blue-400);
+    box-shadow: inset 0 1px 1px rgba(0,0,0,.075), 0 0 8px rgba(0,136,206,.6);
+    transition: box-shadow ease-in-out 0.3s, background-color ease-in 0.3s;
+  }
   &.read-only {
     &:hover {
       .input-field {
-        border: 1px solid transparent;
+        .Border(1px, solid, @color-pf-white);
```
Comment (vikram-raj) [inline src/app/widgets/inlineinput/inlineinput.component.less]: Border color should be transparent.

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2177-9 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2177-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2177-10 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2177-fabric8-planner.badger.fabric8.io

**Comment by SMahil** [comment]
@vikram-raj I didn't get this issue. Can you please tell how to reproduce this issue

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2177-11 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2177-fabric8-planner.badger.fabric8.io

**Comment by vikram-raj** [comment]
@SMahil comment author avatar issue is still there.
screenshot from 2017-10-10 11-36-18

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2177-12 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2177-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2177-13 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2177-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2177-14 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2177-fabric8-planner.badger.fabric8.io

**Comment by rgarg1** [comment]
Not sure why but I cannot open the preview. Can anyone else open it?

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2177-15 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2177-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2177-16 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2177-fabric8-planner.badger.fabric8.io

**Comment by vikram-raj** [comment]
@SMahil In the board view user avatar shrink.

screenshot from 2017-10-15 06-45-47

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2177-17 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2177-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2177-18 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2177-fabric8-planner.badger.fabric8.io

**Comment by rgarg1** [comment]
@SMahil The issue https://github.com/openshiftio/openshift.io/issues/963 asks for larger description and comments fields, but the field sizes appear to be the same as that in preview mode.

**Comment by rgarg1** [comment]
@SMahil The issues mentioned here https://github.com/openshiftio/openshift.io/issues/964 are not fixed. I cannot enter text into description or comment fields without using the Pen icon. Also, saving my changes via Keyboard is not working.

**Comment by rgarg1** [comment]
Escaping edits using keyboard is not working @SMahil

**Comment by rgarg1** [comment]
@SMahil The issue https://openshift.io/openshiftio/openshiftio/plan/detail/1547 still exists.

**Comment by SMahil** [comment]
@rgarg1 As per UX Suggestion I make description fields and comments area consistent rather than making larger.

afa escaping edits using keyboard concerned, I didn't add new functionality enhancement

For entering text into description or comment fields without using the Pen icon, that should markdown component feature which is in patternfly-ng

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2177-19 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2177-fabric8-planner.badger.fabric8.io

**Comment by rgarg1** [comment]
@SMahil Agreed! The original defects which are not addressed in this PRs need to be updated accordingly so that the issues are not lost. Can you pls update the defects referenced?

**Comment by rgarg1** [comment]
@SMahil The issue https://openshift.io/openshiftio/openshiftio/plan/detail/1547 needs a fix.

Also, the comments and description fields are overflowing and need a fix.

comment and desc spillover

**Comment by SMahil** [comment]
@rgarg1 I fixed #1547. Also for overflown content, I have added temporary fix. This has to be fixed in markdown component. Please verify the changes

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2177-20 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2177-fabric8-planner.badger.fabric8.io

**Comment by rgarg1** [comment]
@SMahil Both issues are fixed. Great work. Approving!

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2177-21 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2177-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2177-22 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2177-fabric8-planner.badger.fabric8.io

**Comment by fabric8cd** [comment]
@SMahil fabric8/fabric8-ui:SNAPSHOT-PR-2177-23 is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-2177-fabric8-planner.badger.fabric8.io

**Pull request merged**